README.md

```diff
@@ -65,55 +85,152 @@ antlr4 -Dlanguage=Python3 -visitor /tests/Expr.g4
+Fields and labels of the source `BaseNode` are also available on the `AliasNode`.
+If an `AliasNode` field name collides with these, it takes precedence when accessing that property.
+
+This is what a custom node looks like:
 
 ```python
-from antlr_ast import AstNode
+class NotExpr(AliasNode):
+    _fields_spec = ["expr", "op=NOT"]
+```
+
+This code defines a custom node, `NotExpr` with an `expr` and an `op` field.
+
+##### Field specs
+
+The `_fields_spec` class property is a list that defines the fields the custom node should have.
+
+This is how a field spec in this list is used when creating an custom node from a `BaseNode` (the source node):
 
-class SubExpr(AstNode):
-    _fields = ['expr->expression']
+- If a field spec does not exist on the source node, it is set to `None`
+- If multiple field specs define the same field, the first one that isn't `None` is used
+- If a field spec is just a name, it is copied from the source node
+- If a field spec is an assignment, the left side is the name of the field on the `AliasNode`
+  and the right side is the path that should be taken starting in the source node to get the node
+  that should be the value for the field on the custom node.
+  Parts of this path are separated using `.`
 
-class BinaryExpr(AstNode):
-    _fields = ['left', 'right', 'op']
+##### Connecting to the transformer
 
-class NotExpr(AstNode):
-    _fields = ['NOT->op', 'expr']
+To use this custom node, add a method to the transformer:
+
+```python
+class Transformer(BaseNodeTransformer):
+    # ...
+
+    # here the BaseNode name is the same as the custom node name
+    # but that isn't required
+    def visit_NotExpr(self, node):
+        return NotExpr.from_spec(node)
 ```
 
-Note that `_fields` allows you to redefine rule names.
-For example, in the grammar, the rule for matching expressions is named "expr".
-In this case, using `'expr->expression'` says that the expr field should be called "expression" instead.
+Instead of defining methods on the transformer class to use custom nodes, it's possible to do this automatically:
+
+```python
+Transformer.bind_alias_nodes(alias_nodes)
+```
 
-### Connecting to Visitor
+To make this work, the `AliasNode` classes in the list should have a `_rules` class property
+with a list of the `BaseNode` names it should transform.
 
-Once you have defined your AST classes, you can connect them to the visitor, using their `_from_fields` method.
-For example..
+This is the result:
 
 ```python
-AstVisitor(ExprVisitor):                                  # antlr visitor subclass
-    def visitBinaryExpr(self, ctx):                       # method on visitor
-        return BinaryExpr._from_fields(self, ctx)         # _from_fields method
+class NotExpr(AliasNode):
+    _fields_spec = ["expr", "op=NOT"]
+    _rules = ["NotExpr"]
+
+class Transformer(BaseNodeTransformer):
+    pass
+
+alias_nodes = [NotExpr]
+Transformer.bind_alias_nodes(alias_nodes)
```

Comment:
What happens if I don't subclass this, but do `BaseNodeTransformer.bind_alias_nodes(...)`?

Reply:
It would work as long as the `BaseNodeTransformer` isn't used for multiple languages in the same process.